client/src/unifyfs-sysio.c

```diff
@@ -487,74 +468,60 @@ int UNIFYFS_WRAP(__fxstat)(int vers, int fd, struct stat* buf)
             return -1;
         }
 
-        if (!buf) {
-            errno = EINVAL;
-            return -1;
-        }
-
-        /* get the file id for this file descriptor */
-        int fid = unifyfs_get_fid_from_fd(fd);
-        if (fid < 0) {
-            errno = EBADF;
-            return -1;
-        }
-
-        /* lookup stat info for this file id */
-        int ret = unifyfs_fid_stat(fid, buf);
-        if (ret != UNIFYFS_SUCCESS) {
-            errno = unifyfs_err_map_to_errno(ret);
-            return -1;
-        }
-
-        /* success */
-        return 0;
+        fid = unifyfs_get_fid_from_fd(fd);
+        path = unifyfs_path_from_fid(fid);
+        return __stat(path, buf);
     } else {
         MAP_OR_FAIL(__fxstat);
-        int ret = UNIFYFS_REAL(__fxstat)(vers, fd, buf);
-        return ret;
+        return UNIFYFS_REAL(__fxstat)(vers, fd, buf);
     }
 }
 
 /* ---------------------------------------
  * POSIX wrappers: file descriptors
  * --------------------------------------- */
 
-/* read count bytes info buf from file starting at offset pos,
- * returns number of bytes actually read in retcount,
- * retcount will be less than count only if an error occurs
- * or end of file is reached */
-int unifyfs_fd_read(int fd, off_t pos, void* buf, size_t count,
-                    size_t* retcount)
+/*
+ * Read 'count' bytes info 'buf' from file starting at offset 'pos'.
+ *
+ * Returns number of bytes actually read, or -1 on error, in which
+ * case errno will be set.
+ */
+size_t unifyfs_fd_read(int fd, off_t pos, void* buf, size_t count)
```

Comment:
Perhaps change return value to `ssize_t`, like POSIX read, since it supports -1 values without a cast.

Reply:
Will do